ledger/metrics/src/test/lib/scala/com/daml/metrics/api/testing/TestingInMemoryMetrics.scala

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.metrics.api.testing
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.{AtomicInteger, AtomicLong, AtomicReference}
+
+import com.daml.metrics.api.MetricHandle.{Counter, Factory, Gauge, Histogram, Meter, Timer}
+import com.daml.metrics.api.testing.TestingMetrics.{
+  TestingInMemoryCounter,
+  TestingInMemoryGauge,
+  TestingInMemoryHistogram,
+  TestingInMemoryMeter,
+  TestingInMemoryTimer,
+}
+import com.daml.metrics.api.{MetricHandle, MetricName, MetricsContext}
+import com.daml.scalautil.Statement.discard
+
+import scala.collection.concurrent.{TrieMap, Map => ConcurrentMap}
+
+trait TestingInMemoryMetricsFactory extends Factory {
```

Comment:
Nit: As this object is within a test module and under the `testing` namespace, you may want to remove the `Testing` prefix to improve readability.